### DISCUSSION TITLE
Get null for router objet in portals since 9.5

### DISCUSSION THREAD

**Discussion by antoinelin** [vercel/next.js#15870] [Help]
Hello 👋 

I would like to use a NextJS `Link` inside a React portal, in Next 9.4 it was working without needing to provide any context. In Next 9.5 I got an error telling me that `router` is `null` within the `Link` component. Is there a provider I could import from the core component or someone has any other solution I could use to make is works ? 🙏 

Thank you !

**Comment by mohsen1** [comment]
I'm having the same problem with React Portals. It seems to me the `RouterContext` is set here:
https://github.com/vercel/next.js/blob/cb786ebd81f637b8231e664d9e2dce690e34d5e4/packages/next/client/index.tsx#L511

I tried something like this:

```ts
import singletonRouter, { makePublicRouterInstance } from 'next/router';
import { RouterContext } from 'next-server/dist/lib/router-context';


// ...
 <Portal>
                {singletonRouter.router && (
                  <RouterContext.Provider value={makePublicRouterInstance(singletonRouter.router)}>
                    {children}
                  </RouterContext.Provider>
                )}
</Portal>
```

I'm still seeing the same issue unfortunately. 

@toinelin did you find a solution around this?

**Reply by antoinelin** [reply]
Hi, late answer sorry.
No I did not found the time to get my hands on it yet. Did not try new releases too, I'm gonna test and try to find a solution.

**Reply by antoinelin** [reply]
Still getting this error on `9.5.3`
